usr/share/rear/verify/CDM/default/410_use_replica_cdm_cluster_cert.sh

```diff
@@ -24,13 +24,24 @@ while true; do
     PrintError "Please enter a non-empty CDM cluster IP."
 done
 
+# The name of the tar file that is being downloaded has changed in Rubrik CDM v5.1.
+# Before Rubrik CDM v5.1 it was rubrik-agent-sunos5.10.sparc.tar.gz
+# since Rubrik CDM v5.1 it is rubrik-agent-solaris.sparc.tar.gz
+# cf. https://github.com/rear/rear/issues/2441
+
 CDM_SUNOS_TAR=rubrik-agent-sunos5.10.sparc.tar.gz
+CDM_SOLARIS_TAR=rubrik-agent-solaris.sparc.tar.gz
+CDM_TAR_FILE=$CDM_SUNOS_TAR
 cd /tmp
-/usr/bin/curl $v -kLOJ https://${CDM_CLUSTER_IP}/connector/${CDM_SUNOS_TAR} 
-StopIfError "Could not download https://${CDM_CLUSTER_IP}/connector/${CDM_SUNOS_TAR}"
+/usr/bin/curl $v -fskLOJ https://${CDM_CLUSTER_IP}/connector/${CDM_TAR_FILE} 
+if [[ $? -gt 0 ]];  then
+    CDM_TAR_FILE=$CDM_SOLARIS_TAR
+    /usr/bin/curl $v -fkLOJ https://${CDM_CLUSTER_IP}/connector/${CDM_TAR_FILE} 
+fi
+StopIfError "Could not download Rubrik agent from https://${CDM_CLUSTER_IP}/connector/${CDM_SUNOS_TAR} or https://${CDM_CLUSTER_IP}/connector/${CDM_SOLARIS_TAR}."
```

Comment:
Alternative approach could be to simply give `curl` both URLs and check that you got at least one file. That will make it easier to adapt the code to a future version of CDM that will call the Linux client `rubrik-agent-linux.tar.gz` 😄

Reply:
I will implement something like that tomorrow...

Reply:
Puh!
That became almost a complete rewrite.